src/team/stats.ts

```diff
@@ -0,0 +1,184 @@
+  let bblrPts: number = Math.pow(stats.bblr, 1.5) * 10;
+  const finalPts: number = stats.finals / 120;
+  const bedPts: number = stats.beds / 60;
+  const winPts: number = stats.wins / 30;
+  if (stats.fkdr > 10) {
+    fkdrPts = 0.1 * Math.pow(stats.fkdr, 2) + 2.5 * stats.fkdr + 35;
+  }
+  if (stats.wlr > 10) {
+    wlrPts = 45 * stats.wlr - 134;
+  }
+  if (stats.bblr > 10) {
+    bblrPts = 45 * stats.bblr - 134;
+  }
+  const score =
+    fkdrPts + starPts + wlrPts + bblrPts + finalPts + bedPts + winPts;
+  return score * 10;
+}
+
+export default async function stats(
+  msg: Message,
+  args: TeamArgs
+): Promise<void> {
+  let teams;
+  await msg.react("👍");
+  if (args.splitCommand[2]) {
+    const specifiedTag = args.splitCommandLower[2].toUpperCase();
+    teams = await listTeams({ tag: specifiedTag });
+    if (!teams.length) {
+      await msg.channel.send(
+        errorEmbed()
+          .setTitle("Invalid team tag")
+          .setDescription(`A team with tag \`${specifiedTag}\` doesn't exist.`)
+      );
+      return;
+    }
+  } else {
+    if (!args.teamMembership) {
+      await msg.channel.send(
+        errorEmbed()
+          .setTitle("No team found")
+          .setDescription(
+            "In order to use this command without a team tag, you need to be in a team.\nTry `!team stats TAG`."
+          )
+      );
+      return;
+    }
+    teams = await listTeams({ id: args.teamMembership.team_id });
+  }
+  const members = await listTeamMemberships({ team_id: teams[0].id });
+  const memberIds = members.map((item) => item.user_id);
+  const uuids = [];
+  for (let i = 0; i < memberIds.length; i++) {
+    const uuid = await listUsers({ discord_id: memberIds[i] });
+    uuids.push(uuid[0].minecraft_uuid);
+  }
+  const stats = [];
+  for (let i = 0; i < uuids.length; i++) {
+    const player = await fetchBedwarsData(uuids[i]);
+    stats.push(player);
+  }
+  const initTeamData = {
+    finals: 0,
+    beds: 0,
+    wins: 0,
+    games: 0,
+    fDeaths: 0,
+    bLost: 0,
+    losses: 0,
+    fkdr: 0,
+    bblr: 0,
+    wlr: 0,
+    winPercent: 0,
+    rating: 0,
+    stars: 0,
+  };
+  const teamData = stats.reduce((acc, cur) => {
+    acc.finals += cur.finals;
+    acc.beds += cur.beds;
+    acc.wins += cur.wins;
+    acc.games += cur.games;
+    acc.fDeaths += cur.fDeaths;
+    acc.bLost += cur.bLost;
+    acc.losses += cur.losses;
+    acc.fkdr += cur.fkdr;
+    acc.bblr += cur.bblr;
+    acc.wlr += cur.wlr;
+    acc.stars += cur.stars;
+
+    return acc;
+  }, initTeamData);
+
+  teamData.fkdr /= stats.length;
+  teamData.bblr /= stats.length;
+  teamData.wlr /= stats.length;
+  teamData.winPercent = (teamData.wlr * 100) / (teamData.wlr + 1);
+  teamData.rating = calculateRating(teamData);
+  Object.keys(teamData).map(function (key) {
+    if (!(teamData[key] === Math.round(teamData[key]))) {
+      teamData[key] = parseFloat(teamData[key].toFixed(2));
+    }
+  });
```

Comment:
Oh you're trying to round every number. I'd do something like this:
```suggestion
  const roundedStats = Object.fromEntries(Object.entries(teamData).map([key, value] => value.toFixed(2)))
```

Reply:
The reason why I didn't use that is because I don't want to convert whole numbers to two point fixed ones (e.g. 3 to 3.00) that would display in the embed. (best example is that it's much cleaner to have it be 4400 wins instead of 4400.00 wins)

Reply:
Also I'm trying it out in typescript playground right now and it is a *pain* in the ass to get it to work

Reply:
So for now, I'm gonna keep it like this (if there is like a bot-breaking problem that this causes, I'll gladly fix it)